com.unity.render-pipelines.core/Runtime/Common/CollectionExtensions.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using JetBrains.Annotations;
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// A set of extension methods for collections
+    /// </summary>
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// Removes a range of elements from the list.
+        /// </summary>
+        /// <param name="list">The list to remove the range</param>
+        /// <param name="index">The zero-based starting index of the range of elements to remove</param>
+        /// <param name="count">The number of elements to remove.</param>
+        /// <typeparam name="TList">The type of the input list</typeparam>
+        /// <typeparam name="TValue">The value type stored on the list</typeparam>
+        /// <exception cref="ArgumentOutOfRangeException">index is less than 0. or count is less than 0.</exception>
+        /// <exception cref="ArgumentException">index and count do not denote a valid range of elements in the list</exception>
+        [CollectionAccess(CollectionAccessType.ModifyExistingContent)]
+        public static void RemoveRange<TList, TValue>([DisallowNull] this TList list, int index, int count)
```

Comment:
Can you create as well `Try` version of the API? It will be more useful in our runtime context.

It is still usefull to have the same behaviour as the standard library, but non-throwing API will be good.

To clarifiy, I am thinking of:

```csharp
// Returns `true` on success, with error = null
// Returns `false` on error, with error containing the related exception
public static bool TryRemoveRange<TList, TValue>([DisallowNull] this TList list, int index, int count, [NotNullWhen(false)] out Exception error)
```